src/html/fetch-markdown.js

```diff
@@ -9,137 +9,38 @@
-      Authorization: `token ${token}`,
-    });
-  }
-
-  // compute the URI
-  const uri = computeURI(rootPath, owner, repo, ref, path);
-
-  logger.debug(`fetching file from ${uri}`);
-  const res = await fetch(uri, options);
-  if (!res.ok) {
-    res.options = options;
-    return Promise.reject(res);
-  }
-  // Update sources
-  if (branch && branch !== ref) {
-    setdefault(context.content, 'sources', []).push(computeURI(rootPath, owner, repo, branch, path));
-  } else {
-    setdefault(context.content, 'sources', []).push(uri);
-  }
-  return res;
-}
-
-async function fetchMarkdown(context, { secrets = {}, request, logger }) {
-  const { content } = context;
-  const { path } = request.params;
-
-  try {
-    const res = await fetchFileInRepo(path, context, { secrets, request, logger });
-    content.body = res && await res.text();
-  } catch (err) {
-    if (err.status === 404) {
-      logger.error(`Could not find file at ${err.url}`);
-      setdefault(context, 'response', {}).status = 404;
-    } else if (err.status === 502 || err instanceof TimeoutError) {
-      logger.error(`Gateway timeout of ${secrets.HTTP_TIMEOUT} milliseconds exceeded for ${err.url}`);
-      setdefault(context, 'response', {}).status = 504;
-    } else {
-      logger.error(`Error while fetching file from ${err.url} with the following `
-                  + `options:\n${inspect(err.options, { depth: null })}`);
-      setdefault(context, 'response', {}).status = 502;
-    }
-    if (err instanceof Error) {
-      context.error = err;
-    } else {
-      const msg = await err.text();
-      context.error = new Error(msg);
-    }
-  }
-}
-
-async function fetchAll(context, action) {
+async function fetchMarkdown(context, { request, downloadManager }) {
   setdefault(context, 'content', {});
-
-  const { request, secrets = {}, logger } = action;
   if (!request || !request.params) {
     throw new Error('Request parameters missing');
   }
-
-  if (!secrets.HTTP_TIMEOUT) {
-    logger.warn('No HTTP timeout set, risk of denial-of-service');
-  }
-
-  // get required request parameters
+  const { content } = context;
   const {
-    owner, repo, ref, path,
+    owner, repo, branch, path, ref,
   } = request.params;
 
-  // bail if a required parameter cannot be found
-  if (!owner) {
-    throw new Error('Unknown owner, cannot fetch content');
-  }
-  if (!repo) {
-    throw new Error('Unknown repo, cannot fetch content');
-  }
-  if (!path) {
-    throw new Error('Unknown path, cannot fetch content');
-  }
-  if (!ref) {
-    logger.warn(`Recoverable error: no ref given for ${repo}/${owner}.git${path}, falling back to master`);
-  }
+  try {
+    const res = await downloadManager.fetchGithub({
+      owner,
+      repo,
+      ref,
+      path,
+      errorOn404: true,
+    });
+
+    content.body = res.body;
 
-  return fetchMarkdown(context, action);
+    // Update sources, but use the branch instead of the resolved ref. todo: why ?
```

Comment:
The `context.sources` will be evaluated by https://github.com/adobe/helix-pipeline/blob/87fa02657623338940060296565b8ed1f6ad2a81/src/html/set-surrogate-key.js to set the `Surrogate-Keys` header, which allows selective invalidation.

Reply:
yes. but why do we use the `branch` instead of `ref` ? 

but I think I know, since the `ref` source is stable and never changes....